ecl/hql/hqlutil.cpp

```diff
@@ -2244,6 +2255,21 @@ unsigned isEmptyRecord(IHqlExpression * record)
     return true;
 }
 
+unsigned isSimpleRecord(IHqlExpression * record)
+{
+    ForEachChild(i, record)
+    {
+        IHqlExpression * cur = record->queryChild(i);
+        switch (cur->getOperator())
```

Comment:
Probably a bit restrictive.  It should also support no_record (record inheritance) and no_attr/no_attr_expr (e.g., maxlength)